### PR TITLE
Implement Vibes API

### PR DESCRIPTION
**Pull request by hawkinse**

### CODE REVIEW

File: rwatch/ui/vibes.h [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+/* vibes.h
+ * Provides vibration services declaration.
+ * RebbleOS
+ *
+ * Author: Elliot Hawkins <elliotshawkins@gmail.com>
+ */
+
+#include <stdint.h>
+
+typedef struct VibePattern{
```
Comment (ginge) [inline rwatch/ui/vibes.h]: space on brace please

File: rcore/main.c [hunk 2]
```diff
@@ -16,11 +16,11 @@ int main(void)
     hardware_init();
 
     rebbleos_init();
-    
+
```
Comment (ginge) [inline rcore/main.c]: not sure about touching this file as part of the PR. But I do agree it is messy

**Comment by ginge** [comment]
Looks great, thanks!
A couple of really petty minors in there, but happy to merge.
However, vibrate currently doesn't work on hardware. I think there are some build errors and certainly some null deref somewhere. I would like to fix that before we turn vibe back on.

**Comment by hawkinse** [comment]
I've added the brace spaces and removed changes to main.c. At this point it'll lock up on use due to the commented out vibrate_init(), merging should probably wait until this is addressed.

**Comment by ginge** [comment]
superb work. lovely test too. thanks!

**Pull request merged**